.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,262 @@
+name: Build and Test
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    types: [opened, reopened]
+    branches:
+      - '**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: openjdk:8-jdk
+    env:
+      _JAVA_OPTIONS: "-Xmx3G -Xms2G"
+    defaults:
+      run:
+        working-directory: .
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup System Tools
+        run: |
+          apt update -y
+          apt install -y gnupg2 curl
+      - name: Install Node.js
+        run: |
+          curl -fsSL https://deb.nodesource.com/setup_16.x | bash -
+          apt-get install -y nodejs
+      - name: Check Node.js version
+        run: node --version
+      - name: Verify files
+        run: |
+          curl -sSL https://secchannel.rsk.co/SUPPORT.asc | gpg2 --import -
+          gpg2 --verify SHA256SUMS.asc && sha256sum --check SHA256SUMS.asc
+
+      - uses: actions/cache@v4
+        name: Cache Gradle
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            rskj-core/build
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Build
+        run: |
+          ./configure.sh
+          ./gradlew --no-daemon dependencies
+          ./gradlew --no-daemon --stacktrace build -x test
+
+  sonarqube:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Verify Java installation
+        run: java -version
+
+      - uses: actions/cache@v4
+        name: Restore Gradle cache
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            rskj-core/build
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get Gradle jar
+        run: |
+          if [ ! -f ./gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+          fi
+
+      - name: Run SonarQube analysis
+        run: |
+          sudo apt-get update && sudo apt-get install -yqq git
+          extra_flags=""
+          if [ -n "$CIRCLE_PULL_REQUEST" ]; then
```

Comment:
This particular environment variable isn't going to fly. The `github` context does have some utilities like the ones you used below, so it's better to check for those in here.

Reply:
ah man, how did that sneak back in. no, I know it isn't, mate (as you can see a couple of lines below). will change